esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,798 @@
+        "figure 3 in Carvalhais et al. (2014).", ['mean', 'perc'], ['global'],
+        _get_ancestor_files(diag_config, 'tau_ctotal'))
+
+    with ProvenanceLogger(diag_config) as provenance_logger:
+        for key, value in my_files_dict.items():
+            global_tau_mod[key] = {}
+
+            # load the data
+            ctotal = _load_variable(value, 'ctotal')
+            gpp = _load_variable(value, 'gpp')
+            tau_ctotal = _calc_turnover(ctotal, gpp, key)
+            global_tau_mod['grid'][key] = tau_ctotal
+
+            # apply the GPP threshold and set the data in dictionary
+            gpp_global = gpp.collapsed(['latitude', 'longitude'],
+                                       iris.analysis.SUM)
+            ctotal_global = ctotal.collapsed(['latitude', 'longitude'],
+                                             iris.analysis.SUM)
+            tau_global = ctotal_global / gpp_global
+            tau_global.convert_units('yr')
+
+            global_tau_mod['global'][key] = np.float(tau_global.core_data())
+
+            if diag_config['write_plots']:
+                base_name_mod = (
+                    'global_{title}_{source_label}_'
+                    '{grid_label}'.format(
+                        title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                        source_label=key,
+                        grid_label=diag_config['obs_info']['grid_label']))
+                plot_path_mod = get_plot_filename(base_name_mod, diag_config)
+                # plot_path_list.append(plot_path_mod)
+                _plot_single_map(plot_path_mod, tau_ctotal,
+                                 global_tau_mod['global'][key], key,
+                                 diag_config)
+                provenance_logger.log(plot_path_mod, provenance_record)
+
+        if diag_config['write_netcdf']:
+            model_cubes = [
+                c for c in global_tau_mod['grid'].values()
+                if isinstance(c, iris.cube.Cube)
+            ]
+            obs_cubes = [
+                c for c in global_tau_obs['grid'].values()
+                if isinstance(c, iris.cube.Cube)
+            ]
+            netcdf_path = get_diagnostic_filename(base_name, diag_config)
+            save_cubes = iris.cube.CubeList(model_cubes + obs_cubes)
+            iris.save(save_cubes, netcdf_path)
+        else:
+            netcdf_path = None
+
+        if diag_config['write_plots']:
+            # multimodel agreement
+            base_name_multimodel = '{prefix}_{base_name}'.format(
+                prefix='global_multimodelAgreement', base_name=base_name)
+            plot_path_multimodel = get_plot_filename(base_name_multimodel,
+                                                     diag_config)
+            _plot_multimodel_agreement(plot_path_multimodel, global_tau_mod,
+                                       global_tau_obs, config)
+            provenance_logger.log(plot_path_multimodel, provenance_record)
```

Comment:
This plot is different from the plots above, shoudn't it have a different caption?

Reply:
I am not sure what this exactly means. But, all figures have different captions, handled individually within each plot function.

Reply:
@bouweandela : is there anything more that can/should be changed here?

Reply:
I think so, the caption is stored in the provenance record, but you appear to use the same `provenance_record` for all plots

Reply:
I see. I assumed that a provenance record is per diagnostic and not per plot. Is there a documentation on the conventions on provenance or anything that explains how it's done. I am running blind here.

Reply:
> I assumed that a provenance record is per diagnostic and not per plot.

Provenance is per output file, plot or otherwise, I believe we discussed this earlier here: https://github.com/ESMValGroup/ESMValTool/pull/1395#discussion_r417206222?. 

>  Is there a documentation on the conventions on provenance

Documentation about provenance is a bit scattered, but it is available, for example [here](https://esmvaltool.readthedocs.io/en/latest/api/esmvaltool.diag_scripts.shared.html#esmvaltool.diag_scripts.shared.ProvenanceLogger), [here](https://esmvaltool.readthedocs.io/en/latest/community/diagnostic.html#recording-provenance), and [here](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/interfaces.html#information-provided-by-the-diagnostic-script-to-esmvalcore) and the [example diagnostic](https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/diag_scripts/examples/diagnostic.py
) shows a good example of how to implement it.

Reply:
Thanks. Let me give it a go with the help of the examples.